src/restruct_poc/model/node.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "src/restruct_poc/types.hpp"
+
+namespace openturbine {
+
+struct Node {
+    int ID;      // Node identifier
+    Array_7 x;   // Node positions and orientations
+    Array_7 u;   // Node displacement
+    Array_6 v;   // Node velocity
+    Array_6 vd;  // Node acceleration
+
+    Node(
+        int id, Array_7 position, Array_7 displacement = Array_7{0., 0., 0., 1., 0., 0., 0.},
```

Comment:
`id` should probably be `size_t` or `unsigned_int`.

https://stackoverflow.com/questions/131803/unsigned-int-vs-size-t

Reply:
`id` is an `int` because that's the type used for indexing within the views. Otherwise, I would agree with you. I'm also using an ID of `-1` to denote an invalid node for fixed and prescribed boundary condition constraints. Which definitely needs to be documented better.